### PR TITLE
Javascript javascript2 week2

### PR DESCRIPTION
**Pull request by SathiC1990**
List: Traverse through your array of the currency rate objects, and display them in a grid of items containing all the relevant details.
 Implement a function to find a specific currency rate, searching by the currency from or to.

### CODE REVIEW

File: week 2/script.js [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+class CurrencyRate {
+    constructor(code, rate, date) {
+        this.code = code;
+        this.rate = rate;
+        this.date = date;
+    }
+}
+
+const currencyRatesArray = Object.entries(currencyRates.rates).map(([code, rate, date]) => new CurrencyRate(code, rate, currencyRates.date));
```
Comment (vdaguenet) [inline week 2/script.js:18]: You are making your life complicated here. The array of currency rates could simply be
```js
const currencyRatesArray = [
  {
    timestamp: 1519296206,
    base: "EUR",
    date: "2021-03-17",
    rates: {
        USD: 1.23396
    }
  },
  {
    timestamp: 1519296206,
    base: "DKK",
    date: "2021-03-17",
    rates: {
        SEK: 0.987
    }
  }
];
```

File: week 2/script.js [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+function insertCurrencyRate() {
+    const base = document.getElementById('base').value.toUpperCase();
```
Comment (vdaguenet) [inline week 2/script.js:21]: Don't forget to also `trim()` the base

File: week 2/script.js [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+        alert('Curreny already exists in the list');
+        return;
+    }
+
+
+    currencyRates.timestamp = Date.now();
+    currencyRates.date = new Date().toISOString().split('T')[0];
+    currencyRates.rates[targetCurrency] = rate;
+    document.getElementById('newRateOutput').textContent = JSON.stringify(currencyRates, null, 2);
+  // const select = document.getElementById('targetCurrency').textContent;
```
Comment (vdaguenet) [inline week 2/script.js:54]: Avoid leaving dead code in your PR. I write this comment here, but it also applies to the rest of the file and the other files.

File: week 2/script.js [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+    populateDatalist();
+}
+
+function findCurrencyRate(currencyCode){
+    let selectedArrIndex = currencyRatesArray.findIndex(obj => obj.code == currencyCode);
+    return currencyRatesArray[selectedArrIndex].rate;
+}
+
+function convertRate(from, to){
+    let rate = (to/from).toFixed(3);
```
Comment (vdaguenet) [inline week 2/script.js:120]: This conversion works in some cases but does not handle the conversion from the base currency to one of the rates. See the screenshot below. I should be able to convert 100 EUR (the base currency) to DKK; the result should be 12300. But there is an error in the console. Please have a look at this.


image

File: index.html [hunk 5]
```diff
@@ -1,91 +1,75 @@
...
-            <div calss="text-center2">
-                <p>Exchange rate on: <label id="datetime-label"/></p>
-                
-            </div>
+<html>
+  <head>
+    <link rel="stylesheet" href="styles.css">
+    <title>Currency Conversion</title>
+  </head>
+  <body onload="populateCurrencyDropdown()">
```
Comment (vdaguenet) [inline index.html:7]: Avoid adding binding events with HTML attributes. Prefer using `addEventListener` in JS. I [wrote about that](https://github.com/SathiC1990/HYF-Currency-Converter/pull/1#discussion_r1670895770) last week already.

**Review by vdaguenet — CHANGES_REQUESTED** [review]
I assumed the homework for this week was in the `week 2` folder so I didn't look at the other files. You don't need to create a folder per week. You can instead merge the PR for week 1 when it's approved and then work on the same files.

File: index.html (hunk 5 above)
Comment (SathiC1990) [reply index.html:7]: Done.

File: week 2/script.js (hunk 4 above)
Comment (SathiC1990) [reply week 2/script.js:120]: Done.

**Comment by SathiC1990** [comment]
Hi @vdaguenet I have done the changes for week 2 HW and done the  week-3 homework together. Please review my code.

**Comment by vdaguenet** [comment]
> Hi @vdaguenet I have done the changes for week 2 HW and done the week-3 homework together. Please review my code.

Thanks for letting me know @SathiC1990. I wrote my comments on the PR for week 3, then.

**Pull request merged**